File: src/GameState/IntroState.java
package GameState;

import Main.GamePanel;
import Manager.GameStateManager;
import Manager.KeyManager;
import Manager.LanguageManager;
import Manager.TextureManager;
import java.awt.Color;
import java.awt.Graphics2D;

/**
 * @author Kevin
 */
public class IntroState extends GameState {
    
    private final int INTRO_STATE = 0;
    
    private int alpha;
    private int ticks;

    private final int FADE_IN = 60;
    private final int LENGTH = 60;
    private final int FADE_OUT = 60;
    
    private String lblGame;
    
    public IntroState(GameStateManager gsm) {
        super(gsm);
    }
    
    @Override
    public void init(){
        System.out.println("Iniciando IntroState");
        ticks = 0;
        lblGame = LanguageManager.getInstance().get("nombre_juego");
    }
    
    @Override
    public void update(){
        handleEvents();
        ticks++;
        if(ticks < FADE_IN) {
            alpha = (int) (255 - 255 * (1.0 * ticks / FADE_IN));
            if(alpha < 0) alpha = 0;
        }
        if(ticks > FADE_IN + LENGTH) {
            alpha = (int) (255 * (1.0 * ticks - FADE_IN - LENGTH) / FADE_OUT);
            if(alpha > 255){
                alpha = 255;
            }
        }
        if(ticks > FADE_IN + LENGTH + FADE_OUT) {
            gsm.changeState(new MenuState(gsm));
        }
    }
    
    @Override
    public void draw(Graphics2D g){
        try {
            g.setColor(new Color(255, 255, 255));
            g.fillRect(0, 0, GamePanel.PANEL_WIDTH, GamePanel.PANEL_HEIGHT);
            g.drawImage(
                    TextureManager.logo, 
                    0,
                    0,
                    null
            );
            g.setColor(new Color(0, 0, 0, alpha));
            g.fillRect(0, 0, GamePanel.PANEL_WIDTH, GamePanel.PANEL_HEIGHT);
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
        
    }
    
    @Override
    public void exit(){
        System.out.println("Finalizando IntroState");
    }
    
    @Override
    public void handleEvents(){
        if (KeyManager.isPressed(KeyManager.ENTER)) {
            gsm.changeState(new MenuState(gsm));
        }
    }
    
    @Override
    public int getStateID() {
        return INTRO_STATE;
    }
}

File: src/main/kotlin/app/gamestate/IntroState.kt
package app.gamestate

import app.enums.GameStates
import app.manager.GameStateManager
import app.manager.KeyManager
import app.manager.TextureManager.getImage
import javafx.scene.canvas.GraphicsContext
import javafx.scene.paint.Color
import org.apache.logging.log4j.LogManager

private val logger = LogManager.getLogger(IntroState::class.java)
class IntroState(gameStateManager: GameStateManager) : GameStateBase(gameStateManager) {
    private val FADE_IN = 80
    private val FADE_OUT = 80
    private val LENGTH = 80

    private var ticks = 0
    private var alpha = 1.0

    init {
        initialize()
    }

    override fun initialize() {
        logger.info("Entering IntroState")
    }

    override fun update() {
        handleInput()
        ticks++
        if (ticks < FADE_IN) {
            alpha = 1.0 - 1.0 * ticks / FADE_IN
            if (alpha < 0) alpha = 0.0
        }
        if (ticks > FADE_IN + LENGTH) {
            alpha = 1 * (1.0 * ticks - FADE_IN - LENGTH) / FADE_OUT
            if (alpha > 1) {
                alpha = 1.0
            }
        }
        if (ticks > FADE_IN + FADE_OUT + LENGTH) {
            gameStateManager.changeState(GameStates.MENU)
        }
    }

    override fun draw(graphics: GraphicsContext) {
        graphics.drawImage(getImage("logo"), 0.0, 0.0)
        graphics.fill = Color.rgb(0, 0, 0, alpha)
        graphics.fillRect(0.0, 0.0, 640.0, 640.0)
    }

    override fun handleInput() {
        if (KeyManager.isPressed(KeyManager.K_ENTER)) {
            logger.info("User pressed enter. Skipping intro screen.")
            gameStateManager.changeState(GameStates.MENU)
        }
    }
}